scrapper.py

```diff
@@ -25,39 +39,159 @@ class Crawler:
     """
     Crawler implementation
     """
-    def __init__(self, seed_urls, max_articles: int):
-        pass
+
+    def __init__(self, seed_urls, total_max_articles: int):
+        self.seed_urls = seed_urls
+        self.total_max_articles = total_max_articles
+        self.urls = []
 
     def _extract_url(self, article_bs):
-        pass
+        article_summaries_bs = article_bs.find_all("div", class_="obj_article_summary")
+        for article_summary_bs in article_summaries_bs:
+            link_to_pdf = article_summary_bs.find('a', class_='obj_galley_link pdf')
+            if link_to_pdf:
+                url_link = article_summary_bs.find('div', class_='title').find('a')['href']
+                if len(self.urls) < self.total_max_articles:
+                    self.urls.append(url_link)
 
     def find_articles(self):
         """
         Finds articles
         """
-        pass
+
+        for url in self.seed_urls:
+            response = requests.get(url)
+            sleep_period = random.randint(1, 3)
+            sleep(sleep_period)
+
+            if not response.ok:
+                continue
+
+            soup = BeautifulSoup(response.text, 'lxml')
+            self._extract_url(soup)
 
     def get_search_urls(self):
         """
         Returns seed_urls param
         """
-        pass
+        return self.seed_urls
+
+
+class HTMLParser:
+    def __init__(self, article_url, article_id):
+        self.article_url = article_url
+        self.article_id = article_id
+        self.article = Article(article_url, article_id)
+
+    def _fill_article_with_text(self, article_bs):
+        title = article_bs.find('h1', class_='page_title').text.strip()
+        back_to_seed = article_bs.select_one('nav ol li:nth-child(3) a')['href']
+        seed_bs = BeautifulSoup(requests.get(back_to_seed).text, 'lxml')
+        sections = seed_bs.find_all("div", class_="obj_article_summary")
+        for section in sections:
+            if title in section.text:
```

Comment:
remember: you need to make your code shallow. that will be one of your tasks next Friday if you do not make it earlier.

Reply:
Done